sys/mips/mips/locore.S

```diff
@@ -163,9 +163,18 @@ VECTOR(_locore, unknown)
 	/*
 	 * Enable userspace cycle-count, cycle-count resolution, and CPU
 	 * number registers available via rdhwr.
+	 * Also enable beri specific statcounters.
 	 */
 	MFC0	t0, MIPS_COP_0_HWRENA
-	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4)
+	or	t0, MIPS_HWRENA_CC | MIPS_HWRENA_CCRES | MIPS_HWRENA_CPUNUM | (7 << 4) \
+	| MIPS_BERI_HWRENA_RESET_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_ICACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_DCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2CACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_MEM_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGCACHE_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_L2MASTER_STATCOUNTERS \
+	| MIPS_BERI_HWRENA_TAGMASTER_STATCOUNTERS
```

Comment:
OK.  What you've implemented looks like the right solution in this case (other than we really ought to have a macro for what ever `(7 << 4)` is.)